libr/util/vector.c

```diff
@@ -143,10 +143,13 @@ R_API void r_vector_remove_at(RVector *vec, size_t index, void *into) {
 
 R_API void *r_vector_insert(RVector *vec, size_t index, void *x) {
 	r_return_val_if_fail (vec, NULL);
-	if (vec->len >= vec->capacity) {
+	if (R_MAX (vec->len, index) >= vec->capacity) {
```

Comment:
why this? index >= vec->capacity and you use NEXT_VECTOR_CAPACITY you may end up anyway with a vector that is not big enough for index.

Reply:
damn this shouldnt be commited, this is wrong. ill remove it and fix the CI i pushed the code because i had to run afk. and ill open an issue for all the problems i found in RVector and RPVector